### DISCUSSION TITLE
GitHub Actions Cache both present and not accessible?

### DISCUSSION THREAD

**Discussion by msfjarvis** [community/community#25339] [Actions]
<p>I have a few separate caches set up for my Gradle-based project but I noticed today that cache restoration seems to not happen at all. Actions says there is no cache hit when it attempts to restore it before the build, then says it can’t create the cache because it already exists. I’m not sure what’s up with this, are caches not accessible to PRs from forks?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/98452196fc53d6da015c52b9d3a55de7ba45ea30.png" data-download-href="/uploads/short-url/lJ2GGnctvm0Iyqkr7OkQRpEWqRy.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/13348378/181090508-86eef186-be58-439e-80be-e77e90cf01f0.png" alt="image" data-base62-sha1="lJ2GGnctvm0Iyqkr7OkQRpEWqRy" width="690" height="328" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/98452196fc53d6da015c52b9d3a55de7ba45ea30_2_690x328.png, https://aws1.discourse-cdn.com/github/original/2X/9/98452196fc53d6da015c52b9d3a55de7ba45ea30.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/98452196fc53d6da015c52b9d3a55de7ba45ea30.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/98452196fc53d6da015c52b9d3a55de7ba45ea30_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1005×479 47.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/msfjarvis">@msfjarvis</a>,</p>
<blockquote>
<p>are caches not accessible to PRs from forks?</p>
</blockquote>
<p>Sure, caches is available to PRs from forks. However, there are some restrictions for accessing a cache that you need to know.<br>
According to the description below from the docs about <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache" rel="nofollow noopener">cache</a>:</p>
<blockquote>
<p>With <strong>v2</strong> of the <strong>cache</strong> action, you can access the cache in workflows triggered by any event that has a <strong>GITHUB_REF</strong>.</p>
</blockquote>
<blockquote>
<p>A workflow can access and restore a cache created in the current branch, the base branch (including base branches of forked repositories), or the default branch (usually <strong>master</strong>).</p>
</blockquote>
<p>The cache created for a branch would not be accessible to its base branch and other branches which are not based on this branch.</p>
<p>Similarly, the cache created in the forked repository would not be accessible to the original repository. If you create a cache in the forked repository, this cache is not accessible to the PR opened for merging from the forked repository into the original repository, because this PR is opened in the original repository.</p>

**Comment by msfjarvis** [comment]
<p>Welp, that’d explain it. Thanks for your assistance!</p>

**Comment by mikebourbeauart** [comment]
<p><a class="mention" href="/brightran">@brightran</a> I’ve been having the same problem. I’ve been creating a cache during my release pushes but it’s never found during future releases. According to your explanation that’s because I’m creating the cache on a release branch and not a base branch.</p>
<p>What would the proper workflow be then? Should I run a separate GitHub Action on the master branch to cache my data and only run it again if I need to update the cache? I’m using cache@v2 btw.</p>
